### PR TITLE
Add test for createImageBitmap colorSpaceConversion option - take 2

### PR DESCRIPTION
**Pull request by aardgoose**
This option is currently not tested.
This uses a source image with a wide gamut color space which gives large and easily detectable differences with the two values supported.

The makeBlob factory function has been modified to allow the source image to be changed as required.

### CODE REVIEW

File: html/canvas/element/manual/imagebitmap/createImageBitmap-colorSpaceConversion.html [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        check(1 * width / 4, 1 * height / 4, 255,0,0,255);
+        check(3 * width / 4, 1 * height / 4, 0,249,0,255);
+        check(1 * width / 4, 3 * height / 4, 30,0,239,255);
+        check(3 * width / 4, 3 * height / 4, 0,0,0,255);
+    }
+}
+
+function testDrawImageBitmap(source, options)
+{
+    console.log(source);
```
Comment (miketaylr) [inline html/canvas/element/manual/imagebitmap/createImageBitmap-colorSpaceConversion.html:33]: Can we remove the log please?

File: html/canvas/element/manual/imagebitmap/createImageBitmap-colorSpaceConversion.html [hunk 2]
```diff
@@ -0,0 +1,64 @@
+<!DOCTYPE html>
+<html>
+<title>createImageBitmap + drawImage test</title>
```
Comment (miketaylr) [inline html/canvas/element/manual/imagebitmap/createImageBitmap-colorSpaceConversion.html:3]: nit: should we mention colorSpaceConversion somewhere in the title?

File: html/canvas/element/manual/imagebitmap/createImageBitmap-colorSpaceConversion.html [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    var tolerance = 10; // high tolerance for differing colour management results
+    const check = (x, y, r, g, b, a) =>
+        _assertPixelApprox(canvas, x,y, r,g,b,a, `${x},${y}`, `${r},${g},${b},${a}`, tolerance);
+
+    if (options.colorSpaceConversion == 'none') {
+        check(1 * width / 4, 1 * height / 4, 124,0,27,255);
+        check(3 * width / 4, 1 * height / 4, 0,124,46,255);
+        check(1 * width / 4, 3 * height / 4, 60,0,123,255);
+        check(3 * width / 4, 3 * height / 4, 0,0,0,255);
+    } else {
```
Comment (miketaylr) [inline html/canvas/element/manual/imagebitmap/createImageBitmap-colorSpaceConversion.html:23]: I don't know a ton about canvas, so forgive the naive question, but [the spec states](https://html.spec.whatwg.org/#:~:text=If%20val%20is%20%22default%22%2C%20the%20color%20space%20conversion%20behavior%20is%20implementation%2Dspecific):

> If val is "default", the color space conversion behavior is implementation-specific, and should be chosen according to the default color space that the implementation uses for drawing images onto the canvas.

Should we be asserting implementation-specific behavior in this clause?

Comment (aardgoose) [reply html/canvas/element/manual/imagebitmap/createImageBitmap-colorSpaceConversion.html]: You are right, checks for default option removed.

File: html/canvas/element/manual/imagebitmap/createImageBitmap-colorSpaceConversion.html (hunk 1 above)
Comment (aardgoose) [reply html/canvas/element/manual/imagebitmap/createImageBitmap-colorSpaceConversion.html]: done

File: html/canvas/element/manual/imagebitmap/createImageBitmap-colorSpaceConversion.html (hunk 2 above)
Comment (aardgoose) [reply html/canvas/element/manual/imagebitmap/createImageBitmap-colorSpaceConversion.html]: Title updated

**Review by miketaylr — APPROVED** [review]
LGTM

**Pull request merged**